### PR TITLE
Fix tx expiring soon test

### LINKED ISSUE #3718: Intermittent RPC test failure in p2p_txexpiringsoon.py
https://ci.z.cash/#/builders/12/builds/695

```
=== Running testscript p2p_txexpiringsoon.py ===
  File "/home/zcbbworker/debian9-0/debian9/build/qa/rpc-tests/test_framework/test_framework.py", line 121, in main
    self.run_test()
  File "/home/zcbbworker/debian9-0/debian9/build/qa/rpc-tests/p2p_txexpiringsoon.py", line 299, in run_test
    assert_equal({tx3.hash}, set(self.nodes[2].getrawmempool()))
  File "/home/zcbbworker/debian9-0/debian9/build/qa/rpc-tests/test_framework/util.py", line 362, in assert_equal
    raise AssertionError("(left == right)%s\n  left: <%s>\n right: <%s>" % (message, str(expected), str(actual)))
Initializing test directory /tmp/test7Ghtt4
MiniNode: Connecting to Bitcoin Node IP # 127.0.0.1:11873 using version 170003
MiniNode: Connecting to Bitcoin Node IP # 127.0.0.1:11875 using version 170003
Assertion failed: (left == right)
  left: <set(['3a1cecc0a43363b5a66535d6c608f8740edf148a0f4f6a33d49210a1fe5ee705'])>
 right: <set([])>
Stopping nodes
Cleaning up
Failed
!!! FAIL: p2p_txexpiringsoon.py !!!
```

### PR DESCRIPTION
**Pull request by Eirik0**
Closes #3718

### CODE REVIEW

**Comment by Eirik0** [comment]
This PR attempts to solve intermittent failures in the rpc test `p2p_txexpiringsoon.py`. Much of this PR is refactoring. There was a fair amount of duplicated code in this test and I cleaned it up as I was figuring out what was going on. I was able to reproduce the failure locally, but only about 1/20 times. I think the problem had to do with `testnode2` not syncing properly. To mitigate this I added some extra asserting (for sanity) and some extra calls to `sync_with_ping`. The semantics of the test might be slightly different, but I think that the essence is in tact. I also connected node2 directly to node0  in case the issue was exacerbated by communicating via node1.

**Comment by Eirik0** [comment]
@zkbot try

File: qa/rpc-tests/p2p_txexpiringsoon.py [hunk 1]
```diff
@@ -6,111 +6,81 @@
...
-            time.sleep(sleep_time)
-            timeout -= sleep_time
-            with mininode_lock:
-                if self.last_pong.nonce == self.ping_counter:
-                    received_pong = True
-        self.ping_counter += 1
-        return received_pong
+from test_framework.util import assert_equal, connect_nodes_bi, fail, \
+    initialize_chain_clean, p2p_port, start_nodes, sync_blocks, sync_mempools
+from tx_expity_helper import TestNode, create_transaction
```
Comment (str4d) [inline qa/rpc-tests/p2p_txexpiringsoon.py]: Typo in helper module name: `tx_expiry_helper`

Comment (Eirik0) [reply qa/rpc-tests/p2p_txexpiringsoon.py]: Whoops... Thanks for catching this. This is now fixed.

**Comment by str4d** [comment]
utACK

@zkbot r+

**Pull request merged**